### PR TITLE
feat(material/legacy-core): create legacy-core entry point

### PR DESCRIPTION
**Pull request by mmalerba**
Sets up @angular/material/legacy-core as a location to aggregate legacy themes as components are replaced by their MDC-based versions.

Also establishes @angular/material's dependency on MDC and moves mdc-helpers to @angular/material/core in preparation for moving MDC based components out of experimental

Notable changes:
- The "all-theme" mixins (`mat.all-component-themes`, `mat.all-component-typographies`, `mat.all-component-colors`, and `mat.all-component-densities`) have been split into a legacy (living under material/legacy-core) and non-legacy (living under material/core) version. In addition the MDC version living under `material-experimental/mdc-core` still exists as well. The plan when moving MDC components out of material-experimental is:
   1. The non MDC theme should be moved from core all-themes to legacy-core all-themes
   2. The MDC theme should be added to core all-themes
   3. The mdc-core version of all-themes should continue to include all MDC themes, even those moved out of experimental
- Since `mat.core` includes typography styles, it too has been split into a legacy and non-legacy versions that call out to their corresponding all-typographies mixin.
- Since the prebuilt files are also based on the all-theme mixins, they too have been split into legacy and non-legacy versions
- `mat.mdc-strong-focus-indicators` and related theme mixins have been rolled into their non-MDC counterparts, and the MDC version of the mixins deleted. There's no reason to have them be separate mixins now that they live in the same package

### CODE REVIEW

**Comment by andrewseguin** [comment]
@josephperrott @devversion Heads up that this will be our initial PR to split off the branch madness. We'll wait for your instructions for landing this and getting branches set up.

**Comment by devversion** [comment]
@andrewseguin Is this PR ready? I think we can start setting up the `14.1.x` branch and switching `main` to be `15.0.0-next.0`. Let me know and I can look into setting this up.

**Review by andrewseguin — APPROVED** [review]

File: src/material/legacy-core/legacy-core.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The `@angular/material/core` package contains common components and services used by multiple other
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material/legacy-core/legacy-core.md:1]: I can change it if we want, but just FYI that this PR doesn't really handle docs or examples at all, I figured that would happen in a later pass. Some other options are:
- Just change the file contents to `TODO: documentation` to be very explicit that nobody has done a docs pass on this yet
- Hold this PR back until we finish the docs

**Review by zarend — APPROVED** [review]

File: src/material/legacy-core/legacy-core.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material/legacy-core/legacy-core.md]: Oh I just looked at this file, and it barely says anything anyways. I just updated it to legacy-core for now, and we can do a pass later to see if we want to say anything more about it

**Pull request merged**